### PR TITLE
OpenStack: Adds support for selecting network type to use.

### PR DESCRIPTION
**Pull request by adduarte**
The network type can be defined in the workflow (test), or
any parent chain.
default is OpenshiftSDN

[OSASINFRA-2346](https://issues.redhat.com/browse/OSASINFRA-2346)

### CODE REVIEW

File: ci-operator/step-registry/openstack/conf/generateconfig/openstack-conf-generateconfig-ref.yaml [hunk 1]
```diff
@@ -14,8 +14,10 @@ ref:
     - name: BASE_DOMAIN
     - name: CONFIG_TYPE
       default: "minimal"
-    - name: KURYR_ENABLED
```
Comment (EmilienM) [inline ci-operator/step-registry/openstack/conf/generateconfig/openstack-conf-generateconfig-ref.yaml:17]: is it safe to remove KURYR_ENABLED ?

Comment (adduarte) [reply ci-operator/step-registry/openstack/conf/generateconfig/openstack-conf-generateconfig-ref.yaml:17]: I am pretty sure. but I will check. I don't think is used anywhere.

Comment (adduarte) [reply ci-operator/step-registry/openstack/conf/generateconfig/openstack-conf-generateconfig-ref.yaml:17]: ha good point. looks liek it has been used. we can live it there I guess.

Comment (adduarte) [reply ci-operator/step-registry/openstack/conf/generateconfig/openstack-conf-generateconfig-ref.yaml:17]: The KURYR_ENABLED variable is replaced by NETWORK_TYPE, 
i.e. KURYR_ENABLED=true is the same things as NETWORK_TYPE=Kuryr, 
but there is a minimal danger so lets move the removal to a separate commit.

**Comment by adduarte** [comment]
Before merging this we should make sure all the currently defined workflows execute correctly.

**Comment by pierreprinetti** [comment]
/hold cancel
The failing tests seem to be the "normal" failing tests, most of which because this hasn't merged yet https://github.com/openshift/origin/pull/2613

**Pull request merged**